core/domain/stats_domain.py

```diff
@@ -16,213 +16,256 @@
-        return [cls(answer_log_model.answers)
-                for answer_log_model in answer_log_models]
+(stats_models,) = models.Registry.import_models([models.NAMES.statistics])
 
-    @classmethod
-    def get(cls, exploration_id, state_name, rule_str):
-        """Gets the StateRuleAnswerLog domain object corresponding to the
-        given rule spec string and state_name.
 
-        Args:
-            exploration_id: str. The id of the exploration.
-            state_name: str. The name of the state whose answer log is to be
-                fetched.
-            rule_str: str. A string representation of a rule. Only the answers
-                that match this rule are returned.
-
-        Returns:
-            StateRuleAnswerLog. The answer log domain object corresponding to
-                the given exploration id, state name, and rule spec string.
-        """
-        # TODO(sll): Deprecate this method.
-        return cls.get_multi_by_multi_explorations(
-            [(exploration_id, state_name)], [rule_str])[0]
-
-    def get_all_top_answers(self):
-        """Returns all the answers sorted by count.
-
-        Returns:
-            list(tuple). A list of 2-tuples in the form (answer, count) sorted
-                by count. `answer` is a string that represents a answer
-                submitted to that state of the exploration. `count` of type int,
-                is the number of times that answer has been given.
-        """
-        return sorted(
-            self.answers.iteritems(), key=operator.itemgetter(1),
-            reverse=True)
-
-    def get_top_answers(self, num_answers_to_return):
-        """Returns the top `num_answers_to_return` answers.
-
-        Args:
-            num_answers_to_return: the maximum number of answers to return.
-
-        Returns:
-            list(tuple). A list of (answer, count) tuples for the
-                `num_answers_to_return` answers with the highest counts.
-                `answer` is a string that represents a answer submitted to that
-                state of the exploration. `count` of type int, is the number of
-                times that answer has been given.
-        """
-        return self.get_all_top_answers()[:num_answers_to_return]
+# These are special sentinel values attributed to answers migrated from the old
+# answer storage model. Those answers could not have session IDs or time spent
+# values inferred or reconstituted perfectly, so they are assigned these
+# values, instead. Logic and jobs which use these values are expected to skip
+# over the migrated answers to avoid tainted results. Furthermore, all migrated
+# answers are easy to retrieve by reducing session value on this session ID.
+# NOTE TO DEVELOPERS: All other state answer data model entities must not ever
+# store this session ID unless it was created by the AnswerMigrationJob. Also,
+# this string must never change.
+MIGRATED_STATE_ANSWER_SESSION_ID = 'migrated_state_answer_session_id_2017'
```

Comment:
The idea was to keep this general since we don't actually know whether we'll perform another answer migration. Keeping 2017 in the label makes sense in case we do another migration, but the variable can trivially be renamed if we do perform another migration in the future. That being said, I'm quite ambivalent about this so I did the change anyway.